### PR TITLE
Separate and Update Fuchsia

### PR DESCRIPTION
**Pull request by cramertj**
The first commit pulls Fuchsia out into its own directory, while the second commit updates the "open" flags as in #848.

cc @smklein, @alexcrichton

### CODE REVIEW

**Comment by cramertj** [comment]
@alexcrichton Should I open a PR against rustc updating the submodule, or is there a more official process for updating libc?

**Comment by alexcrichton** [comment]
Nah that's it!

File: src/lib.rs [hunk 1]
```diff
@@ -286,6 +286,9 @@ cfg_if! {
     } else if #[cfg(target_os = "redox")] {
         mod redox;
         pub use redox::*;
+    } else if #[cfg(target_os = "fuchisa")] {
```
Comment (petrhosek) [inline src/lib.rs]: There's a typo: `fuchisa`, so this is likely not going to work.

Comment (cramertj) [reply src/lib.rs]: Gah! https://github.com/rust-lang/libc/pull/854

**Pull request merged**